src/libraries/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.T.cs

```diff
@@ -424,14 +424,7 @@ public void MultipleCallsToGetSpan()
                 Assert.True(span.Length >= 256);
                 Span<T> newSpan = output.GetSpan();
                 Assert.Equal(span.Length, newSpan.Length);
-
-                unsafe
-                {
-                    void* pSpan = Unsafe.AsPointer(ref MemoryMarshal.GetReference(span));
-                    void* pNewSpan = Unsafe.AsPointer(ref MemoryMarshal.GetReference(newSpan));
-                    Assert.Equal((IntPtr)pSpan, (IntPtr)pNewSpan);
-                }
-
+                Assert.Equal(0, Unsafe.ByteOffset(ref MemoryMarshal.GetReference(span), ref MemoryMarshal.GetReference(newSpan)));
```

Comment:
`Unsafe.AreSameRef`?

Reply:
I can do `Unsafe.AreSame`, but I figured this might be better for debugging if it does fail. Will see what others' opinions are, but can change it to that if that's what we want.

Reply:
@jkotas what's your opinion about this?

Reply:
Either way is fine with me. I do not have a strong opinion.

> I figured this might be better for debugging if the test does fail.

I think it is very unlikely that this test is ever going to fail. We can always add more verbose diagnostic if we see the test failing intermittently.